### PR TITLE
Pubstwo 1255

### PR DESCRIPTION
**Pull request by mbucknell**
Implemented filtering based on a search term. Also did some styling cleanup on the app. Please note that I have now defaulted the development server to run in multi-threaded mode.

### CODE REVIEW

File: pubs_ui/manager/static/js/views/SearchView.js [hunk 1]
```diff
@@ -1,54 +1,52 @@
...
 	'views/BaseView',
 	'views/AlertView',
 	'hbs!hb_templates/search',
 	'backgrid',
-	'select-all',
-	'paginator',
-	'backbone.stickit',
-	'models/PublicationCollection'
-], function (bootstrap, Handlebars, BaseView, AlertView, hbTemplate, Backgrid, SelectAll, Paginator, Stickit, PublicationCollection) {
+	'backgrid-select-all',
```
Comment (mbucknell) [inline pubs_ui/manager/static/js/views/SearchView.js]: Note that I changed paginator and select-all. to backgrid-paginator and backgrid-select-all to make it cleared that these are extensions of backgrid.

File: pubs_ui/manager/static/js/views/SearchView.js [hunk 2]
```diff
@@ -1,54 +1,52 @@
...
+			var $pubList;
 
-			this.$('.loading-indicator').show();
+			BaseView.prototype.render.apply(this, arguments);
+			$pubList = this.$('.pub-grid');
 
 			// Set the elements for child views and render if needed.
 			this.alertView.setElement(this.$('.alert-container'));
 
-			//Don't render grid until the publications have been fetched.
```
Comment (mbucknell) [inline pubs_ui/manager/static/js/views/SearchView.js]: I opted to render the grid immediately so at least the titles are visible. The loading indicator will show when the pubs have been fetched.

File: pubs_ui/manager/static/js/views/SearchView.js [hunk 3]
```diff
@@ -56,91 +54,112 @@ define([
...
-				cell: "string"
-			}, {
-				name: "title",
-				label: "  Title",
-				editable: false,
-				cell: "string"
-			}];
+			// Set up collection event handlers and then fetch the collection
+			this.listenTo(this.collection, 'backgrid:selected', this.editPublication);
+			this.listenTo(this.collection, 'request', this.showLoadingIndicator);
```
Comment (mbucknell) [inline pubs_ui/manager/static/js/views/SearchView.js]: These last two automatically handle the showing and hiding of the load indicator so when something like getFirstPage is called which ultimately calls fetch, the loading indicator visibility is set appropriately.

File: pubs_ui/manager/templates/manager/manager.html [hunk 4]
```diff
@@ -69,8 +69,8 @@
 				'backbone' : "{{ url_for('bower.static', filename='backbone/backbone.js') }}",
 				'backbone.paginator' : "{{ url_for('bower.static', filename='backbone.paginator/lib/backbone.paginator.js') }}",
 				'backgrid' : "{{ url_for('bower.static', filename='backgrid/lib/backgrid.js') }}",
-				'select-all' : "{{ url_for('bower.static', filename='backgrid-select-all/backgrid-select-all.js') }}",
-				'paginator' : "{{ url_for('bower.static', filename='backgrid-paginator/backgrid-paginator.js') }}",
+				'backgrid-select-all' : "{{ url_for('bower.static', filename='backgrid-select-all/backgrid-select-all.js') }}",
```
Comment (mbucknell) [inline pubs_ui/manager/templates/manager/manager.html]: Prepended backgrid- to select-all and paginator to make it clear these are backgrid extensions.

File: run.py [hunk 5]
```diff
@@ -12,6 +12,6 @@
         host = host_val
     else:
         host = '127.0.0.1'
-    application.run(host=host, port=5050)
+    application.run(host=host, port=5050, threaded=True)
```
Comment (mbucknell) [inline run.py]: Here is where I have enabled the development server to run in multi threaded mode.

**Comment by mbucknell** [comment]
customJasmineRequireConfig.txt is the only one that I have set up to use spaces. I'll change this to tabs tomorrow.

**Comment by mamcderm-usgs** [comment]
:+1: Thanks for the comments, very helpful.

**Comment by mbucknell** [comment]
Added a couple of tests to PublicationCollection to test using filters.

**Pull request merged**